packages/components/src/mobile/ui-strings/index.native.js

```diff
@@ -0,0 +1,12 @@
+const stringsToAdd = [];
+
+export const addStrings = ( newStrings ) => stringsToAdd.unshift( newStrings );
+
+const getStrings = () =>
+	stringsToAdd.reduce( ( previous, current ) => {
+		return { ...previous, ...current() };
+	}, {} );
+
+export const withUIStrings = ( WrappedComponent ) => ( props ) => (
```

Comment:
I like this HOC a lot! 😀 

I agree with @guarani 's [comment](https://github.com/WordPress/gutenberg/pull/25238#discussion_r494701487) about it being a bit unclear that "UiStrings" are actually just strings that need to come from native though. Maybe something like "withNativeStrings" or withStringsFromNative" would be clearer. Or perhaps a comment here would be a better way to make it clear. Obviously I don't have a great solution here. 😞

Reply:
I wasn't sure about this name either :)

The rationale was to imply that these strings are Text displayed on UI elements to the final user.

Even though it's declared on `mobile` components, I guess this could be used by web if needed (there's nothing `Native` on this). They might have a different way of approaching these needs though. I'm not sure.

I was also talking with @Tug about finding a `better/more-gutenberg` way of doing this, so I wouldn't worry too much since this might change soon.

Reply:
Do we expect a lot more strings to come from native? In that case I do understand this approach. For now I'm not that much in favor of it though as it's a native specific HOC and it will just make our life harder when we'll try to make x-platform components

Reply:
> Do we expect a lot more strings to come from native?

So far I don't really think so. Things like this will be use only if we need to show text not allowed to be committed on this repo. This shouldn't happen very often.

As mentioned, I'll test the filter approach and make the change if it works as expected on a upcoming PR.